### PR TITLE
[deckhouse] move ModuleConfig handling to the deckhouse module

### PR DESCRIPTION
**Pull request by yalosev**
## Description
1. Service `deckhouse` selector removed. EndpointSlice is created manually via the global webhook
1. Module `003-deckhouse-config` has been removed
1. CRD `ModuleConfig` moved to the `002-deckhouse` module
1. `ModuleConfig` sync hooks moved to the `002-deckhouse` module
1. ValidationWebhookConfigurations moved to the `002-deckhouse` module

## Why do we need it, and what problem does it solve?
Extra module `003-deckhouse-config` and external validation webhook image don't known anything about external modules. And we couldn't reload modules configs and settings smoothly.

## What is the expected result?
We don't need an extra deckhouse-config pod restart when changing an external module spec

## Checklist
- [x] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: deckhouse
type: feature
summary: Move ModuleConfig handling to the deckhouse module.
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by yalosev** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 5891223784 5891223784-1-con-1-25 /sys/deckhouse-oss/install:pr5566 centos@3.69.10.23

**Comment by z9r5** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 5890472581 5890472581-1-con-1-25 /sys/deckhouse-oss/install:pr5566 centos@3.78.75.225

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 5890472581 5890472581-1-con-1-25 /sys/deckhouse-oss/install:pr5566 centos@3.78.75.225

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

File: global-hooks/create_endpoints.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	discv1 "k8s.io/api/discovery/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	OnStartup: &go_hook.OrderedConfig{Order: 1},
+}, generateDeckhouseEndpoints)
+
+func generateDeckhouseEndpoints(input *go_hook.HookInput) error {
```
Comment (nabokihms) [inline global-hooks/create_endpoints.go:35]: Why do we need this hook? Could you please elaborate in the comment? There was a discussion that lasted for a week, we definitely need to pass this knowledge to new generations of developers!

File: modules/002-deckhouse/hooks/pkg/apis/validate_deckhouse_cm.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	kwhhttp "github.com/slok/kubewebhook/v2/pkg/http"
+	"github.com/slok/kubewebhook/v2/pkg/model"
+	kwhmodel "github.com/slok/kubewebhook/v2/pkg/model"
+	kwhvalidating "github.com/slok/kubewebhook/v2/pkg/webhook/validating"
```
Comment (nabokihms) [inline modules/002-deckhouse/hooks/pkg/apis/validate_deckhouse_cm.go:29]: I remember @RomanenkoDenys complaining about this library. Could you please contact him and ask which library he thinks fits best.

File: modules/002-deckhouse/hooks/pkg/apis/validation_webhooks.go
Comment (nabokihms) [inline modules/002-deckhouse/hooks/pkg/apis/validation_webhooks.go:1]: Why validating webhooks are in the `pkg/apis` folder? Should it be the `pkg/apis/validation` (like in [kubernetes](https://github.com/kubernetes/kubernetes/tree/master/pkg/apis/apps/validation))?

File: modules/002-deckhouse/hooks/pkg/apis/validate_deckhouse_cm.go (hunk 2 above)
Comment (nabokihms) [reply modules/002-deckhouse/hooks/pkg/apis/validate_deckhouse_cm.go:29]: Or persuade him using the `slok/kubewebhook` library 🙁

**Review by nabokihms — CHANGES_REQUESTED** [review]

File: modules/002-deckhouse/hooks/pkg/apis/validate_deckhouse_cm.go (hunk 2 above)
Comment (yalosev) [reply modules/002-deckhouse/hooks/pkg/apis/validate_deckhouse_cm.go]: Ok, but this library is used by us before. In some images, deckhouse-config and so more

Comment (yalosev) [reply modules/002-deckhouse/hooks/pkg/apis/validate_deckhouse_cm.go]: No problems here. It was another case. We have discussed it and `slok/kubewebhook` is not a blocker

**Pull request merged**